Apps/FlightSoftware/Watchdog/src/main.c

```diff
@@ -219,6 +228,23 @@ int main(void) {
             /* clear event when done */
             loop_flags ^= FLAG_TIMER_TICK;
         }
+
+        if (loop_flags & FLAG_I2C_GAUGE_READING_ACTIVE) {
+            I2C_Sensors__spinOnce();
+
+            I2C_Sensors__Status stat = I2C_Sensors__getGaugeReadingStatus(&i2cReadings);
+
+            int done = (I2C_SENSORS__STATUS__INCOMPLETE != stat);
+
+            if (done) {
+                /* check for kicks from devices and reset misbehaving things */
+                send_earth_heartbeat(i2cReadings);

```

Comment:
We're doubling up on heartbeats right now, since we also send the earth heartbeat on line 220. I personally would rather send the old data in the heartbeat before collecting the new data, as that way we don't have to wait a potentially indeterminate amount of time for the I2C sensors to respond.

Reply:
Makes sense to me, although this does mean we will potentially be sending several heartbeats in a row with the same data before we get new data. Is that preferable to less frequent heartbeats?

Reply:
Sounds good - now just have those two function calls in the switch statement on `rovstate` and nothing in the `if(done)` (outside of removing the `FLAG_I2C_GAUGE_READING_ACTIVE` flag)